hathor/indexes/addresses_rocksdb_index.py

```diff
@@ -0,0 +1,174 @@
+
+
+class AddressesRocksDBIndex(AddressesIndex):
+    """ Index of inputs/outputs by address.
+
+    This index uses rocksdb and the following key format:
+
+        key = [address][tx.timestamp][tx.hash]
+              |--34b--||--4 bytes---||--32b--|
+
+    It works nicely because rocksdb uses a tree sorted by key under the hoods.
+
+    The timestamp must be serialized in big-endian, so ts1 > ts2 implies that bytes(ts1) > bytes(ts2),
+    hence the transactions are sorted by timestamp.
+    """
+    def __init__(self, db: 'rocksdb.DB', *, cf_name: Optional[bytes] = None,
+                 pubsub: Optional['PubSubManager'] = None) -> None:
+        self._db = db
+        self._cf_name = cf_name or _CF_NAME_ADDRESS_INDEX
+        self._cf = self._db.get_column_family(self._cf_name)
+        self.pubsub = pubsub
+        if self.pubsub:
+            self.subscribe_pubsub_events()
+
+    def _to_key(self, address: str, tx: Optional[BaseTransaction] = None) -> bytes:
+        import struct
+        assert len(address) == 34
+        key = address.encode('ascii')
+        if tx:
+            assert tx.hash is not None
+            assert len(tx.hash) == 32
+            key += struct.pack('>I', tx.timestamp) + tx.hash
+            assert len(key) == 34 + 4 + 32
+        return key
+
+    def _from_key(self, key: bytes) -> Tuple[str, int, bytes]:
+        import struct
+        assert len(key) == 34 + 4 + 32
+        address = key[:34].decode('ascii')
+        timestamp: int
+        (timestamp,) = struct.unpack('>I', key[34:38])
+        tx_hash = key[38:]
+        assert len(address) == 34
+        assert len(tx_hash) == 32
+        return address, timestamp, tx_hash
+
+    def subscribe_pubsub_events(self) -> None:
+        """ Subscribe wallet index to receive voided/winner tx pubsub events
+        """
+        assert self.pubsub is not None
+        # Subscribe to voided/winner events
+        events = [HathorEvents.STORAGE_TX_VOIDED, HathorEvents.STORAGE_TX_WINNER]
+        for event in events:
+            self.pubsub.subscribe(event, self.handle_tx_event)
+
+    def _get_addresses(self, tx: BaseTransaction) -> Set[str]:
+        """ Return a set of addresses collected from tx's inputs and outputs.
+        """
+        assert tx.storage is not None
+        addresses: Set[str] = set()
+
+        def add_address_from_output(output: 'TxOutput') -> None:
+            script_type_out = parse_address_script(output.script)
+            if script_type_out:
+                address = script_type_out.address
+                addresses.add(address)
+
+        for txin in tx.inputs:
+            tx2 = tx.storage.get_transaction(txin.tx_id)
+            txout = tx2.outputs[txin.index]
+            add_address_from_output(txout)
+
+        for txout in tx.outputs:
+            add_address_from_output(txout)
+
+        return addresses
+
+    def publish_tx(self, tx: BaseTransaction, *, addresses: Optional[Iterable[str]] = None) -> None:
```

Comment:
The same as `_get_addresses` method. This is exactly the same as in the other index class.

Reply:
I'm moving this to the base class.